src/math_utils.rs

```diff
@@ -25,6 +25,15 @@ pub fn isqrt(n: &BigInt) -> Result<BigInt, VirtualMachineError> {
     Ok(x)
 }
 
+/// Performs integer division between x and y; fails if x is not divisible by y.
+pub fn safe_div(x: &BigInt, y: &BigInt) -> Result<BigInt, VirtualMachineError> {
+    if !(x % y).is_zero() || y.is_zero() {
+        Err(VirtualMachineError::SafeDivFail(x.clone(), y.clone()))
+    } else {
+        Ok(x / y)
+    }
+}
```

Comment:
In fact, you may want to also perform the division only once:
```suggestion
pub fn safe_div(x: &BigInt, y: &BigInt) -> Result<BigInt, VirtualMachineError> {
    if y.is_zero() {
        return Err(VirtualMachineError::DividedByZero);
    }
    let (q, r) = x.div_rem(y);
    if !r.is_zero() {
        return Err(VirtualMachineError::SafeDivFail(x.clone(), y.clone()));
    }
    Ok(q)
```

Reply:
resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/333/commits/bff8fe8a880eec8430a711e8508ccbaee3da8247